posts/2024-11-19-24.0.0.12-beta.adoc

```diff
@@ -0,0 +1,240 @@
+----
+
+The value of `portName` is the port name of the Web Service implementation you're configuring.  This name comes from your `@WebService(portName=<web service port name>` annotated class.
+Alternatively, you can check the `<wsdl:port ... name="Web Service Port Name">` line in your WSDL file for the port name.
+
+
+The following example shows how to enable XML schema validation for a specific client service:
+
+[source,xml]
+----
+<webServiceClient serviceName="<client service name>"  enableSchemaValidation="true" />
+----
+
+The value of `serviceName` is the name of the Web Service Client you're configuring. This name comes from your `@WebServiceClient(serviceName=<client service name>` annotated stub class for managed clients.
+For unmanaged clients, you can check the `<wsdl:service name="<client service name>"` line in your WSDL file.
+
+
+
+=== Default validation
+
+You can configure the default level of JAXB validation of inbound SOAP Messages with the `enableDefaultValidation=true` attribute. Default validation is much more efficient than XML schema validation, so enabling it provides basic message validation with lower overhead. Disabling it lets you ignore various unmarshalling errors for problematic messages. Default validation is enabled by default for web services, but disabled for web service clients.
+
+==== Global default validation
+
+The following example shows how to enable default validation globally for web services in your Open Liberty runtime:
+
+[source,xml]
+----
+<webService enableDefaultValidation="true"/>
+----
+
+Default validation is enabled by default for web service clients. To disable default validation globally for web service clients, set the `enableDefaultValidation="false"` attribute on the `webServiceClient` element.
+
+[source,xml]
+----
+<webServiceClient enableDefaultValidation="false"/>
+----
+
+==== Targeted default validation
+
+The following example shows how to enable default validation for a specific web service:
+
+[source,xml]
+----
+<webService  portName="SayHelloService" enableDefaultValidation="true"/>
+----
+
+Default validation is enabled by default for web service clients. To disable default validation for a specific web service client, set the `enableDefaultValidation="false"` attribute on the `webServiceClient` element and use the `serviceName` attribute to specify the client service.
+
+[source,xml]
+----
+<webServiceClient serviceName="<client service name>"  enableDefaultValidation="false" />
+----
+
+=== Ignore unexpected elements
+
+The new Open Liberty `ignoreUnexpectedElements` attribute provides the same function. Inbound SOAP messages often contain extra elements in the SOAP body when a web service is updated but the client is not. When a message contains an unknown element, Open Liberty throws a `UnmarshallingException: Unknown Element`. By enabling `ignoreUnkownEllements`, you can keep validation enabled while ignoring unknown elements.
```

Comment:
> The new Open Liberty `ignoreUnexpectedElements` attribute provides the same function.

Provides the same function as what?

```suggestion
The new Open Liberty `ignoreUnexpectedElements` attribute provides the same function. Inbound SOAP messages often contain extra elements in the SOAP body when a web service is updated but the client is not. When a message contains an unknown element, Open Liberty throws a `UnmarshallingException: Unknown Element`. By enabling `ignoreUnexpectedElements`, you can keep validation enabled while ignoring unknown elements.
```

Reply:
"same function..." was an artifact from an earlier draft- deleted. Typo addressed in https://github.com/OpenLiberty/blogs/pull/4151